### PR TITLE
Get metadata and namespace of User

### PR DESCRIPTION
**Pull request by baixinsui**
fixed https://github.com/eclipse-xpanse/xpanse/issues/792
related https://github.com/eclipse-xpanse/xpanse/pull/821

### CODE REVIEW

File: src/components/oidc/OidcConfig.ts [hunk 1]
```diff
@@ -33,6 +32,9 @@ export function getRolesOfUser(oidcUserInfo: object): string[] {
 
     if (availableRoleList.length === 0) {
         availableRoleList.push('user'); // default role when no roles are assigned.
+    } else {
+        const userNamespace: string = getMetadataByKey(oidcUserInfo, 'namespace');
+        console.log('namespace of user:', userNamespace);
```
Comment (swaroopar) [inline src/components/oidc/OidcConfig.ts:37]: @baixinsui we are only logging the user namespace here? Do we really need to fetch the 'namespace'?

**Review by swaroopar — CHANGES_REQUESTED** [review]
comment added.

File: src/components/oidc/OidcConfig.ts [hunk 2]
```diff
@@ -20,11 +21,9 @@ export const OidcConfig: OidcConfiguration = {
 export function getRolesOfUser(oidcUserInfo: object): string[] {
     let availableRoleList: string[] = [];
     if ('urn:zitadel:iam:org:project:roles' in oidcUserInfo) {
-        if (
-            typeof oidcUserInfo['urn:zitadel:iam:org:project:roles'] === 'object' &&
-            oidcUserInfo['urn:zitadel:iam:org:project:roles'] !== null
-        ) {
-            const roleList: string[] = Object.keys(oidcUserInfo['urn:zitadel:iam:org:project:roles']);
+        const grantedRolesObject = oidcUserInfo['urn:zitadel:iam:org:project:roles'];
```
Comment (swaroopar) [inline src/components/oidc/OidcConfig.ts:24]: can we also move all the keys names of oidcUserInfo to constants file pls?
Pls consider other fields too in this file.

File: src/components/oidc/OidcConfig.ts (hunk 1 above)
Comment (baixinsui) [reply src/components/oidc/OidcConfig.ts]: The UI  need not to get 'namespace'. I  deleted related code.

File: src/components/oidc/OidcConfig.ts (hunk 2 above)
Comment (baixinsui) [reply src/components/oidc/OidcConfig.ts]: Ok.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**